src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/OneStatementPerLineCheck.java

```diff
@@ -193,7 +193,7 @@ private void checkIfSemicolonIsInDifferentLineThanPrevious(DetailAST ast) {
             int countOfSemiInCurrentLambda = countOfSemiInLambda.pop();
             countOfSemiInCurrentLambda++;
             countOfSemiInLambda.push(countOfSemiInCurrentLambda);
-            if (!inForHeader && countOfSemiInCurrentLambda > 1
+            if (!inForHeader
```

Comment:
Regression will confirm as I don't see how to create a violation on `>=` and 0 will never happen.

Reply:
Regression: http://rveach.no-ip.org/checkstyle/regression/reports/138/

It found EXACTLY 1 case, so I am updating the check and the UTs for this.